packages/dds/tree/src/util/utils.ts

```diff
@@ -471,3 +471,26 @@ export function capitalize<S extends string>(s: S): Capitalize<S> {
 export function compareStrings<T extends string>(a: T, b: T): number {
 	return a > b ? 1 : a === b ? 0 : -1;
 }
+
+/**
+ * Transforms the provided input with the provided transform function, caching the result in the provided map.
+ * Subsequent calls to this function with the same input and cache will return the cached result.
+ *
+ * @param input - The input data to transform.
+ * @param cache - The map in which the transformation output will be cached.
+ * @param transform - The transformation to apply to the input data. Will only be run if no output is already cached.
+ */
+export function transformWithWeakMapCache<TIn extends object, TOut>(
+	input: TIn,
+	cache: MapGetSet<TIn, TOut>,
+	transform: (input: TIn) => TOut,
+): TOut {
+	const cached = cache.get(input);
```

Comment:
Is this function basically just the same thing as `getOrCreate`?

Reply:
...It sure is... Let's just use that then...

Reply:
`getOrAddInMapLazy` at least, yes. Convergent evolution FTW. Removed.

Reply:
Er, doesn't `getOrCreate` do the same thing? Not a big deal, but we should prefer `getOrCreate` since it's part of the top level utils, whereas `getOrAddInMapLazy` is more of a utility for `NestedMap`. 

`getOrAddInMapLazy` should probably be removed.

Reply:
Updated